app/logic/characters.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+from app.logic.belongings import Armor, Weapon, Potion, Effect
+
+
+class Knight:
+
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: list[Armor],
+            weapon: Weapon,
+            potion: list[Potion]
+    ) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.dressed_up = []
+        self.power += self.weapon.power
+
+    def put_on_armour(self) -> None:
+        if self.armour:
+            for item in self.armour:
+                self.put(item)
+
+    def apply_potions(self) -> None:
+        if self.potion:
+            for item in self.potion:
+                self.apply_effect(item.effect)
+
+    def initialize(self) -> None:
+        self.put_on_armour()
+        self.apply_potions()
+
+    def battle(
+            self,
+            rival: Knight
+    ) -> None:
+        self.hp -= (rival.power - self.protection)
+        rival.hp -= (self.power - rival.protection)
+        if self.hp < 0:
+            self.hp = 0
+        if rival.hp < 0:
+            rival.hp = 0
+
+    @staticmethod
+    def init_armor(
+            armor_list: list
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]